### ISSUE TITLE
Profile text displayed right to left but correct on web

### ISSUE THREAD

**Issue by nyankoframe** [tuskyapp/Tusky#3294]
The profile text for https://mastodon.social/dungeons changes based on the current character. At the moment the profile in website looks like this:

![Screenshot_20230210_101352](https://user-images.githubusercontent.com/69328423/218166141-04c98a19-7294-4f54-bb06-002c37072345.jpg)

But it appears in Tusky like this:

![Screenshot_20230210_101059](https://user-images.githubusercontent.com/69328423/218166003-b7e3ad61-f1da-4f25-b578-1188e75aff3b.jpg)

The timeline posts mentioning this character also appear using right to left text:

![Screenshot_20230210_101437](https://user-images.githubusercontent.com/69328423/218166370-2846b6dc-ae08-47b0-b165-89b310c9a657.jpg)

* * * *
- Tusky Version: 21.0
- Android Version: 12
- Android Device: Galaxy S10
- Mastodon instance (if applicable):

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by nikclayton** [comment]
The account is https://mastodon.social/api/v1/accounts/109354069933492315, if you pull that and look at the bytes for the profile info you get:

```
00000000  3c 70 3e e2 9a 94 ef b8  8f 20 e1 8e af ef ac a1  |<p>...... ......|
00000010  f0 9d 98 ad ea ad bf ef  ac a1 2c 20 74 68 65 20  |.........., the |
00000020  44 72 61 67 6f 6e 62 6f  72 6e 20 46 69 67 68 74  |Dragonborn Fight|
00000030  65 72 3c 62 72 20 2f 3e  0a                       |er<br />.|
```

If you paste the text in to an editor that supports bidirectionality (e.g., Android Studio) and run the cursor through the name text you'll see a marker at the top of the cursor swap direction as you move through the text.

The sword emoji is `e2 9a 94 ef b8 8f`, `e1 8e af ef ac a1 f0 9d 98 ad ea ad bf ef  ac a1` is the name. 

I thought this might be https://www.w3.org/International/articles/inline-bidi-markup/#nomarkup, but it doesn't look like there's embedded unicode to change the directionality of the text, it's the individual "letters" themselves.

https://www.babelstone.co.uk/Unicode/whatisit.html seems to confirm that, reporting:

```
U+13AF : CHEROKEE LETTER HI
U+FB21 : HEBREW LETTER WIDE ALEF
U+1D62D : MATHEMATICAL SANS-SERIF ITALIC SMALL L
U+AB7F : CHEROKEE SMALL LETTER HI
U+FB21 : HEBREW LETTER WIDE ALEF
```

Hebrew is written right-to-left (unlike Cherokee).

Given https://developer.android.com/reference/kotlin/androidx/compose/ui/text/style/TextDirection and other references in the docs I think Android is seeing the first Hebrew character as having a strong text direction (right to left) and is using that for the rest of the text.

If I've read it properly this seems to be https://issuetracker.google.com/issues/145943288, so is an Android platform bug.

**Comment by nikclayton** [comment]
Spot checked Megalodon and Fedilab, they handle this correctly, so there is something that can be done.

**Comment by nyankoframe** [comment]
The profile page has changed since I submitted this issue, so here is a link to a post with the same characters if you need it:

https://mastodon.social/@dungeons/109850440310010558

**Comment by nikclayton** [comment]
Hmm.

Did that post have the same problem 3 days ago? I just went to look at it, and I see it showing correctly now.

![image](https://user-images.githubusercontent.com/773100/219431045-2bffa060-eb6c-43bd-b605-609bd6922461.png)

I'll keep poking and see if I can trigger the problem.

**Comment by nikclayton** [comment]
Yeah, I can duplicate this on my profile page. That should be enough to debug some more.

**Comment by nikclayton** [comment]
I think I've tracked this down.

`layout/activity_account.xml` has `android:textDirection="anyRtl"` set on a view close to the root.

Per https://developer.android.com/reference/android/view/View.html#attr_android:textDirection this will set the text direction like so:

> The paragraph direction is RTL if it contains any strong RTL character, otherwise it is LTR if it contains any strong LTR characters. If there are neither, the paragraph direction is the view’s resolved layout direction.

This was added in https://github.com/tuskyapp/Tusky/pull/964.

The discussion on that PR has these two comments:

---

@charlag: Didn't we have this discussion at https://github.com/tuskyapp/Tusky/issues/878 and decided against anyRTL?

@connyduck: yes but https://github.com/tuskyapp/Tusky/issues/878 was about user generated content and this is about our interface (or am I missing something?)

---

If that (setting `anyRtl` on user generated content is bad) correct (I'm not qualified to judge if it is or not) then I think the correct fix on `AccountActivity` is to restore the default text direction (`firstStrong`) on views in `AccountActivity` that show user generated content.

That's the private note, the account bio, and the titles/content of the account fields.

I'll prep a PR that does that for further discussion.
